docs/fold.md

```diff
@@ -0,0 +1,126 @@
+## AIR: Fold instruction
+Here is the `fold` syntax.
+
+`(fold <iterable> <iterator> <instruction> [<last_instruction>])`
+
+Iterable can be: a stream, map, canonicalized stream, canonicalized map or scalar array.
+
+If `last_instruction` is not set, then it's assumed:
+ - `null` in case of `(fold (seq ...`
+ - `never` in case of `(fold (par ...`
+
+Here is a simple example when `fold` loops over a scalar to populate map using `iterator` as key-value pair value:
+```clojure
+(fold iterable iterator
+	(seq
+	  (ap ("key" iterator) %map)
+	  (next iterator)
+	)
+	;; here is an implicit (null)
+)
+```
+
+This example demonstrates the case when `next` is not the last instruction. This feature is supported with `fold`` over scalar only.
+```clojure
+(fold iterable iterator
+	(seq
+	  (next iterator)
+	  (ap ("key" iterator) %map)
+	)
+	;; here is an implicit (null)
+)
+```
+
+Fold body can have no `next`. The body will be executed only once in this case.
```

Comment:
I think it is important to emphasize that this is true for `fold` over scalars and canons only